//
// Copyright 2020 The Feast Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto3";
package feast.core;


option go_package = "github.com/feast-dev/feast/go/protos/feast/core";
option java_outer_classname = "FeatureProto";
option java_package = "feast.proto.core";

import "feast/types/Value.proto";

message FeatureSpecV2 {
    // Name of the feature. Not updatable.
    string name = 1;

    // Value type of the feature. Not updatable.
    feast.types.ValueType.Enum value_type = 2;

    // Tags for user defined metadata on a feature
    map<string,string> tags = 3;
}
